core/src/main/kotlin/org/eln2/sim/electrical/mna/component/Component.kt

```diff
@@ -116,16 +130,18 @@ abstract class Component : IDetail {
 		if (circuit == null) return
 		val n = nodes[nidx].node
 		val tn = nr.node
-		if (tn.mergePrecedence(n) > n.mergePrecedence(tn)) {
-			nodes[nidx].node = nr.node
-		} else {
-			nr.node = nodes[nidx].node
-		}
+		val (nnamed, tnnamed) = Pair(n.nameSet, tn.nameSet)
+		if(nnamed && !tnnamed) tn.named(n.name)
+		if(tnnamed && !nnamed) n.named(tn.name)
+		tn.unite(n)
+		nodes[nidx].node = tn
+
+		// Assertion intended--fail loudly if circuit mutated here.
```

Comment:
I'm thinking that it would be better if we handled this more gracefully; we're talking a full halt of an entire server over one bad circuit here.

Reply:
It's not really a "bad circuit" as "totally illegal program state". This should never happen, and if it does, the error is intentional.

Making it catchable is a good compromise.

Reply:
Make it so.